### ISSUE TITLE
Availability of MassIVE-KB dataset

### ISSUE THREAD

**Issue by irleader** [Noble-Lab/casanovo#131]
From the preprint, "MassIVE-KB contains a designated subset of 30,506,973 “high quality” PSMs, identified by applying a strict (∼0%) PSM-level FDR filter and then selecting at most 100 PSMs for each combination of peptide sequence and charge."

Did you download the original MassIVE knowledge base from https://massive.ucsd.edu/ProteoSAFe/static/massive-kb-libraries.jsp? Which specific library did you download from, is it Human HCD Spectral Library of v1?

Is it possible to share your processing pipeline (simply using PEAKS?) or your processed MassIVE-KB dataset? I would like to train Casanovo by myself. Thanks.

**Comment by bittremieux** [comment]
Rather than a direct download of the MassIVE-KB v1 in MGF format, which only contains a single MS/MS spectrum per precursor (peptidoform), we used up to the top 100 PSMs per precursor. As this is a big dataset, unfortunately these spectra are not readily available as a single download. We did extract all spectra into a single MGF, but it is 131GB big, so it will take multiple days to share this.

To compile the full training dataset yourself, on the [MassIVE website](https://massive.ucsd.edu/ProteoSAFe/static/massive.jsp), go to [MassIVE Knowledge Base](https://massive.ucsd.edu/ProteoSAFe/static/massive-kb-libraries.jsp) > [Human HCD Spectral Library](https://massive.ucsd.edu/ProteoSAFe/status.jsp?task=82c0124b6053407fa41ba98f53fd8d89) > [All Candidate library spectra](https://massive.ucsd.edu/ProteoSAFe/result.jsp?task=82c0124b6053407fa41ba98f53fd8d89&view=candidate_library_spectra) > Download. This will give you a zipped TSV file with the metadata and peptide identifications for all 30 million PSMs. Using the filename (column "filename") you can then retrieve the corresponding spectra from the MassIVE FTP server and extract the required spectra using their scan number (column "scan").

We did not use PEAKS. The MassIVE-KB PSMs were obtained during the initial compilation of MassIVE-KBTTg, as described by [Wang et al.](https://doi.org/10.1016/j.cels.2018.08.004).

**Comment by wsnoble** [comment]
Wout, can you add a section to the [readthedocs](https://casanovo.readthedocs.io/en/latest/) called something like "Datasets," where we can add the above information as well as information about accessing the nine-species benchmark?

**Comment by irleader** [comment]
Hi Wout,

Thanks a lot for the instruction!

"Using the filename (column "filename") you can then retrieve the corresponding spectra from the MassIVE FTP server and extract the required spectra using their scan number (column "scan")."

I guess I have to write a script to download all the required mzML/mzxML files from MassIVE FTP server to local disk first (which might be much larger 131G), then extract required spectra from local files. Am I right?

**Comment by bittremieux** [comment]
Here is the annotated MGF with all 30M PSMs from MassIVE-KB that were used for training: https://filesender.belnet.be/?s=download&token=73843f2a-5964-4695-bbc2-835f1b58121d

This link will be active for a month.

**Comment by irleader** [comment]
Thanks a lot! I have started downloading.

**Comment by yangshu729** [comment]
The annotated MGF file with all 30M PSMs from MassIVE-KB seems to be unavailable since the link is no longer working. Any chance you could share it again? I’d like to retrain the model. Thanks!

**Comment by wsnoble** [comment]
Note that you might want to consider training using multiple enzymes. The 30M set is only tryptic.  You can find training data associated with [this paper](https://pubs.acs.org/doi/full/10.1021/acs.jproteome.4c00422). The data is on [zenodo](https://zenodo.org/records/12587317).  This is the data used to train the currently released version of Casanovo.

Here is the relevant sentence: "Specifically, we trained a standard 9-layer version of Casanovo on a combination of our multienzyme training set (896 373 PSMs from 46 different enzymes) and 2 000 000 tryptic PSMs randomly selected from MassIVE-KB. We selected this training set size based on empirical results suggesting that Casanovo’s test set performance plateaus when trained on increasingly larger subsets of the MassIVE-KB data. [(12)]"

I think the file you want is mskb_final.train.mgf.

**Comment by yangshu729** [comment]
Thank you for the information and for pointing me to the Zenodo dataset. I understand that the current Casanovo model was trained on a mix of multienzyme and tryptic PSMs, and I appreciate the reasoning behind the selected training set.

However, I’m specifically interested in the full 30M PSMs dataset from MassIVE-KB. Due to limited compute resources, I’d like to use this dataset for some quick validation. Would it still be possible to obtain or download this data?

**Comment by bittremieux** [comment]
The MassIVE-Kb data are now also available as annotated MGF files on Zenodo:
- [MassIVE-KB v1 30 million PSMs](https://doi.org/10.5281/zenodo.14973856)
- [MassIVE-KB v1 30 million PSMs training/validation/test splits](https://doi.org/10.5281/zenodo.14967861)
